lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2438,6 +2442,39 @@ void ModuleEmitter::emitRTLModule(RTLModuleOp module) {
       addName(module.getArgument(port.argNum), name);
   }
 
+  // Get the list of reserved words we need to avoid.  We could prepopulate this
+  // into the used words cache, but it is large and immutable, so we just query
+  // it when needed.
+  auto &reservedWords = getReservedWords();
+  auto moduleName = module.getName();
+  if (reservedWords.count(moduleName)) {
+    // If not available, we need to auto-unique it.
+    SmallVector<char, 16> nameBuffer(moduleName.begin(), moduleName.end());
+    nameBuffer.push_back('_');
+    auto baseSize = nameBuffer.size();
+
+    // Try until we find something that works.
+    while (1) {
+      auto suffix = llvm::utostr(nextGeneratedNameID++);
+      nameBuffer.append(suffix.begin(), suffix.end());
+      moduleName = StringRef(nameBuffer.data(), nameBuffer.size());
+
+      if (!reservedWords.count(moduleName)) {
+        auto insertResult = usedModuleNames.insert(moduleName);
+        if (insertResult.second) {
+          // If the module name is unique, then the insertion is successfull,
+          // then the return bool is true, then exit from the infinite while
+          // loop.
+          break;
+        }
+      }
+      // Chop off the suffix and try again.
+      nameBuffer.resize(baseSize);
+    }
+    // Rename the module to a unique non-keyword name.
+    module.setName(moduleName);
```

Comment:
This code should be factored with the code in addName rather than duplicating it.

Reply:
E.g., I'm doing almost the same thing here: https://github.com/llvm/circt/pull/691/files#diff-0ad6cf1c85d571a808d05866900e638ab97439c724412ce8e14ecfcabba44438R287.

Reply:
Added a function to resolve the conflict with keywords.
https://github.com/llvm/circt/pull/697/files#diff-c81b46d284b442266f20fe3cd2a1c7001b3b6f4841da6b3627b092ef4ec3d407R422